src/figuresClasses.ts

```diff
@@ -1,19 +1,104 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    this.color = color;
+
+    const sides = [a, b, c];
+    const longestSide: number = Math.max(...sides);
+    const indexOfLongest = sides.indexOf(longestSide);
+    const otherSides: number[] = sides.slice(0, indexOfLongest);
+    const arePositive: boolean = sides[0] > 0 && sides[1] > 0 && sides[2] > 0;
+    const isTriangle: boolean = longestSide < otherSides[0] + otherSides[1];
+
+    if (!arePositive || !isTriangle) {
+      throw new Error('Not a triangle');
+    }
+
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
[CODE KNOWLEDGE] - When you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to do it manually.
![image](https://github.com/mate-academy/ts_figures_classes/assets/63243136/bb2672b4-e4f1-4b38-8046-c2d3c52f3314)

The same as for `this.color = color;`